client/components/filters/advanced/index.js

```diff
@@ -127,9 +99,15 @@ class AdvancedFilters extends Component {
 
 	addFilter( key, onClose ) {
 		const filterConfig = this.props.config[ key ];
-		const newFilter = { key, rule: filterConfig.rules[ 0 ] };
+		const newFilter = { key };
+		if ( filterConfig.rules ) {
```

Comment:
Do we need to ensure this is an array that has length here so we can access the object safely below?